server/products.js

```
@@ -0,0 +1,44 @@
+const db = require('APP/db')
+const Product = db.model('product')
+
+module.exports = require('express').Router()
+
+  .param('id', (req, res, next, id) => {
+    Product.findbyId(id)
+      .then(foundProduct => {
+        if (!foundProduct) throw Error('omg error 404')
+        req.product = foundProduct
+        next()
+      })
+      .catch(next)
+  })
+
+  .get('/:id', (req, res, next) => {
+    res.send(req.product)
+  })
+
+
+  .get('/', (req, res, next) => {
+    Product.findAll().then(allProducts => res.send(allProducts)).catch(next)
```

to discuss: if the query returns no items, should we throw an error?